server/utils.py

```diff
@@ -253,3 +254,23 @@ def generate_number_table(num):
     Used in models.Assignment.mysql_course_submissions_query
     """
     return ' UNION '.join('SELECT {} as pos'.format(i) for i in range(1, num + 1))
+
+def upload_file(file, container=None, name=None, prefix=None, overwrite=False):
```

Comment:
This method should probably create an `ExternalFile` object too to track the file, instead of having to remember to do that at each call site